internal/cmdrender/executor.go

```diff
@@ -292,6 +292,8 @@ func hydrate(ctx context.Context, pn *pkgNode, hctx *hydrationContext) (output [
 	return output, err
 }
 
+const resourceIDAnnotation = "internal.config.k8s.io/resource-id"
```

Comment:
Why are we creating a new one? Can we use `config.k8s.io/id`?

Reply:
pl. add a comment explaining how is this annotation used ?

Reply:
Updated.